### PR TITLE
Always run android integration tests on emulator

### PR DESCRIPTION
**Pull request by rolandkakonyi**
## Description

Currently when executing `yarn integration-test test:android` it runs on the first available device or emulator.
The integration tests can only run on emulators right now.

## Changes

Added a lookup and auto-start of emulator, to enable passing device ID to set target for running.
Starting the emulator is needed as otherwise we don't know the device ID for it.

## Checklist
- [x] 🗒 `CHANGELOG` entry - internal change, no entry needed

### CODE REVIEW

File: integration_test/package.json [hunk 1]
```diff
@@ -6,9 +6,9 @@
...
-    "stop-test:ios": "xcrun simctl list devices available -e -j | jq --raw-output '.devices.[].[] | select( .state == \"Booted\" and (.deviceTypeIdentifier | contains(\"iPhone\"))).udid' | xargs -I UDID xcrun simctl terminate UDID com.bitmovin.player.reactnative.IntegrationTest 2> /dev/null; exit 0",
-    "start-test:android": "yarn cavy run-android",
+    "stop-test:android": "adb shell am force-stop com.bitmovin.player.reactnative.integrationtest; true",
+    "stop-test:ios": "xcrun simctl list devices available -e -j | jq --raw-output '.devices.[].[] | select( .state == \"Booted\" and (.deviceTypeIdentifier | contains(\"iPhone\"))).udid' | xargs -I UDID xcrun simctl terminate UDID com.bitmovin.player.reactnative.IntegrationTest 2> /dev/null; true",
+    "start-test:android": "yarn cavy run-android --deviceId \"$([ -z \"$(adb devices | grep -v List | grep device | grep -m 1 emulator | cut -f 1)\" ] && (emulator -list-avds | head -n 1 | xargs -I NAME emulator -avd NAME 1> /dev/null 2> /dev/null & adb wait-for-device shell 'while [[ -z $(getprop sys.boot_completed) ]]; do sleep 1; done;'); (adb devices | grep -v List | grep device | grep -m 1 emulator | cut -f 1))\"",
```
Comment (rolandkakonyi) [inline integration_test/package.json:11]: This checks if there is an emulator running, if not starts one, then runs on the running emulator.

File: integration_test/package.json [hunk 2]
```diff
@@ -6,9 +6,9 @@
     "android": "react-native run-android",
     "ios": "react-native run-ios",
     "start": "react-native start",
-    "stop-test:android": "adb shell am force-stop com.bitmovin.player.reactnative.integrationtest",
-    "stop-test:ios": "xcrun simctl list devices available -e -j | jq --raw-output '.devices.[].[] | select( .state == \"Booted\" and (.deviceTypeIdentifier | contains(\"iPhone\"))).udid' | xargs -I UDID xcrun simctl terminate UDID com.bitmovin.player.reactnative.IntegrationTest 2> /dev/null; exit 0",
-    "start-test:android": "yarn cavy run-android",
+    "stop-test:android": "adb shell am force-stop com.bitmovin.player.reactnative.integrationtest; true",
```
Comment (zigavehovec) [inline integration_test/package.json:9]: What does the "true" do?

File: integration_test/package.json (hunk 1 above)
Comment (zigavehovec) [reply integration_test/package.json:11]: Quite a script, would it be possible to extract it into a command with a descriptive name by chance? Something like: "run-on-active-emulator"?

Comment (rolandkakonyi) [reply integration_test/package.json:11]: I wanted to keep the CLI interface minimal, I don't want to have `start-test:ios` and `start-test:android-run-on-active-emulator`.
Wanted to keep it consistent.
Once the limitation is lifted we can remove this or extract IMO

File: integration_test/package.json (hunk 2 above)
Comment (rolandkakonyi) [reply integration_test/package.json:9]: It returns with success exit code even if the previous command failed. E.g. if no emulator was running.

File: integration_test/package.json (hunk 1 above)
Comment (zigavehovec) [reply integration_test/package.json:11]: Makes sense!

**Review by zigavehovec — APPROVED** [review]
LGTM!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -205,7 +205,7 @@ The `package.json` file contains various scripts for common tasks:
 - `yarn example android`: run the example app on Android.
 - `yarn example pods`: install pods only.
 - `yarn integration-test start`: start the Metro server for the integration tests.
-- `yarn integration-test test:android`: run the player tests on Android.
-- `yarn integration-test test:ios`: run the player tests on iOS.
+- `yarn integration-test test:android`: run the player tests on Android emulator.
+- `yarn integration-test test:ios`: run the player tests on iOS simulator.
```
Comment (rolandkakonyi) [inline CONTRIBUTING.md:209]: @zigavehovec I updated the documentation here!

File: integration_test/README.md [hunk 4]
```diff
@@ -24,4 +24,4 @@ yarn integration-test test:android # Run tests on Android emulator
 
 To set the license key to be used for the tests, you can set the key `"licenseKey"` in `integration_test/app.json`.
 
-Note: The tests are currently only supported on iOS simulators and Android emulators. Running them on real devices is not supported at the moment.
+Note: The tests are currently only supported on iOS simulators and Android emulators. Running them on real devices is not supported at the moment because the testing framework has hard-coded `localhost` as the server address.
```
Comment (rolandkakonyi) [inline integration_test/README.md:27]: @zigavehovec I updated the documentation here!

**Pull request merged**